### DISCUSSION TITLE
Add Vulkan Renderer

### DISCUSSION THREAD

**Discussion by AkiraKusanagi** [EmiyaSyahriel/CrossLauncher#87] [Ideas]
If this is possible, it would be nice. Maybe this could be a feature in the Settings menu.

**Comment by EmiyaSyahriel** [comment]
All of the Launcher UI is drawn using Android Canvas API, and the only part that can be changed by developer is should the canvas rendered using hardware acceleration or not. 

The hardware acceleration of Canvas API is set to use OpenGL ES by Android internally and AFAIK cannot be changed by third-party developers _(but maybe the vendor can, idk, maybe some vendor already used Vulkan)_ . Therefore **the main Launcher UI cannot use Vulkan as it's renderer** .

The only part of this project that can be changed to use Vulkan maybe is the wave live wallpaper (and 3D visualizers that is still stub CMake project at this point), which is rendered using OpenGL ES, even then, there is no benefit on adding Vulkan to the live wallpaper since what it render is less than 10k triangles per frame (including the **Dynamic** particles) which is far too light even for OpenGL ES, and might instead introduces overhead and incompatibility problem.

~or maybe I should just rebuilt the entire launcher using C++ (Vulkan AFAIK only available on C/C++) instead of Kotlin?~
